common/src/primitives/time.rs

```diff
@@ -17,48 +17,52 @@
 #![allow(unused, dead_code)]
 use lazy_static::lazy_static;
 use logging::log;
-use std::sync::atomic::{AtomicI64, Ordering};
-use std::time::SystemTime;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::time::{Duration, SystemTime};
+
+fn duration_to_int(d: &Duration) -> Result<u64, Box<dyn std::error::Error>> {
```

Comment:
Out of curiosity: why a boxed error?

Reply:
Btw, this is the idiomatic way to make generic errors in rust. I'm not sure this idea is already on the table before answering before.